src/main/scala/org/deepdive/ddlog/DeepDiveLogDeltaDeriver.scala

```diff
@@ -31,46 +31,51 @@ object DeepDiveLogDeltaDeriver{
       )
     }
 
-    var incCqBodies = new ListBuffer[List[Atom]]()
-    var incCqConditions = new ListBuffer[Option[Cond]]()
+    var incCqBodies = new ListBuffer[List[Body]]()
     // New incremental bodies
-    cq.bodies zip cq.conditions foreach { case (body, cond) =>
+    cq.bodies foreach { bodies =>
+      // We don't support deriving delta rules for modifiers
+      bodies foreach {
+        case x: ModifierAtom => throw new RuntimeException("Deriving delta rules for modifier atom is not supported!")
```

Comment:
Using `collect` below seems like a good idea to keep the code simple, because there's a check here.  However, if someone happens to add a fourth class, they may get silently ignored by the rest of the code, right?  So, I'd suggest writing this guard in a safer way as below:

```scala
case _: Atom | _: Cond => // supported
case _ => throw new RuntimeException ...
```

I think you can use `sys.error("...")` for throwing the same exception.